tests/scale/README.md

```diff
@@ -0,0 +1,398 @@
+# Scale Tests
+
+## Setup
+
+- Create a GKE Cluster using the following details as a guide:
+  - 4 n2d-standard-8 nodes
+  - 32 vCPUs
+  - 128 GB total memory
+  - us-west2-b
+  - 1.27.5-gke.200
+
+- Install Gateway API Resources:
+
+  ```console
+  kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v0.8.1/standard-install.yaml
+  ```
+
+- Install edge NGF and save the Pod Name and LoadBalancer IP for tests:
+
+  ```console
+  helm install scale-test oci://ghcr.io/nginxinc/charts/nginx-gateway-fabric  --create-namespace --wait -n nginx-gateway --version=0.0.0-edge
+  ```
+
+  ```console
+  export NGF_IP=$(kubectl get svc -n nginx-gateway scale-test-nginx-gateway-fabric --output jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  export NGF_POD=$(kubectl get pods -n nginx-gateway -l "app.kubernetes.io/name=nginx-gateway-fabric,app.kubernetes.io/instance=scale-test" -o jsonpath="{.items[0].metadata.name}")
+  ```
+
+- Install Prometheus:
+
+  ```console
+  kubectl apply -f prom-clusterrole.yaml
+  helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+  helm repo update
+  helm install prom prometheus-community/prometheus --set useExistingClusterRoleName=prometheus -n prom
+  ```
+
+### Run the tests
+
+#### Scale Listeners to Max of 64
+
+Test Goal: Measure how NGF performs as the number of Listeners increases to the max of 64.
+
+Test Plan:
+
+- Scale up to 64 HTTP Listeners
+- All Listeners are on a single Gateway
+- Each Listener has 1 HTTPRoute attached
+- Each HTTPRoute references 1 unique Service
+- Services and Deployments are created before scaling Listeners.
+- After each Listener + HTTPRoute is created, measure the time it takes to get a successful response from the new
+  route (time to ready).
+- Record the time to ready in seconds in a csv file for each iteration.
+
+Total Resources Created:
+
+- 1 Gateway with 64 Listeners
+- 64 HTTPRoutes
+- 64 Services, Deployments, Pods
+
+Follow the steps below to run the test:
+
+- Run the test:
+
+  ```console
+  go test -v -run TestScale_Listeners -i 64
+  ```
+
+- [Collect and Record Metrics](#collecting-metrics).
+
+- Cleanup:
+
+  Delete resources from cluster:
+
+  ```console
+  kubectl delete -Rf TestScale_Listeners
+  ```
+
+  Delete generated manifests:
+
+  ```console
+  rm -rf TestScale_Listeners
+  ```
+
+- Check for any errors or restarts after cleanup.
+- Check nginx conf to make sure it looks correct.
```

Comment:
nit: Should we use `NGINX` instead of `nginx`? Looked through Michael's Longevity PR and some other docs and it looks like they use `NGINX`.

Reply:
Sure. Doesn't hurt to be consistent. I'll change.